metals/src/main/scala/scala/meta/internal/metals/ServerLivenessMonitor.scala

```diff
@@ -64,10 +66,12 @@ class ServerLivenessMonitor(
 )(implicit ex: ExecutionContext) {
   private val state: AtomicReference[ServerLivenessMonitor.State] =
     new AtomicReference(ServerLivenessMonitor.Idle)
+  private val isInitialized = new AtomicBoolean(false)
```

Comment:
OR maybe we could include `requestMonitor` in LauncherConnection instead of Monitor and construct ServerLivenessMonitor in BuildServerConnection with the reconnect method already initialized?

Reply:
The first version doesn't really work, since I'd have to use something like `getAndUpdate` and according to docs it shouldn't have any side effects since it may be executed more than once. So I can't create the task there.